compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPULinkExecutables.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/LLVMGPU/Passes.h"
+#include "iree/compiler/Codegen/Utils/LinkingUtils.h"
+#include "iree/compiler/Utils/ModuleUtils.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir::iree_compiler {
+
+#define GEN_PASS_DEF_LLVMGPULINKEXECUTABLESPASS
+#include "iree/compiler/Codegen/LLVMGPU/Passes.h.inc"
+
+namespace {
+
+static bool isAddressSpacePrivate(uint32_t addressSpace) {
+  return addressSpace == /*local*/ 3 || addressSpace == /*private*/ 5;
```

Comment:
~~I think `3` is shared (what we call workgroup) and `5` is local (what we call private)~~

Gah ok as I was writing this I found that NVPTX and AMDGPU do not agree on naming: https://llvm.org/docs/NVPTXUsage.html#address-spaces and https://llvm.org/docs/AMDGPUUsage.html#address-spaces

Based on below usage it looks like `Private` in the function naming is referencing the symbol visibility of the global (?) can you add a comment clarifying that? I'm not really sure what to do about the `/**/` annotations, maybe we just use our own names.

Reply:
Of course they don't :/
It's an LLVM/MLIR issue that they don't use an enum or have any kind of structural definition.

Reply:
worst part is that local == workgroup on nvidia and local == private on amd.